### PR TITLE
Fix internal error issue in debugger

### PR DESCRIPTION
**Pull request by praveennadarajah**
Fixes #https://github.com/ballerina-platform/ballerina-lang/issues/24716

### CODE REVIEW

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java [hunk 1]
```diff
@@ -75,7 +75,7 @@
 
         // TODO: validate file path
         ArrayList<String> command = new ArrayList<>(ballerinaExec);
-        boolean debugTests = args.get("debugTests") != null && Boolean.parseBoolean((String) args.get("debugTests"));
+        boolean debugTests = args.get("debugTests") != null && (boolean) args.get("debugTests");
```
Comment (NipunaRanasinghe) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java:78]: Please avoid this unsafe casting of client args (`object` -> `boolean`)

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java [hunk 2]
```diff
@@ -90,7 +90,7 @@
 
         command.add(balFile);
 
-        boolean networkLogs = args.get("networkLogs") != null && Boolean.parseBoolean((String) args.get("networkLogs"));
+        boolean networkLogs = args.get("networkLogs") != null && (boolean) args.get("networkLogs");
```
Comment (NipunaRanasinghe) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java:93]: Please avoid this unsafe casting of client args (`object` -> `boolean`)

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java (hunk 1 above)
Comment (praveennadarajah) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java]: Fixed

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java (hunk 2 above)
Comment (praveennadarajah) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java]: Fixed

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java [hunk 3]
```diff
@@ -90,18 +95,29 @@
...
             command.add("--b7a.http.tracelog.host=localhost");
             command.add("--b7a.http.tracelog.port=" + networkLogsPort.intValue());
         }
 
-        ArrayList<String> commandOptions = (ArrayList<String>) args.get("commandOptions");
+        ArrayList<String> commandOptions = null;
+        if (args.get("commandOptions") instanceof ArrayList) {
+            commandOptions = (ArrayList<String>) args.get("commandOptions");
+        }
         commandOptions = commandOptions == null ? new ArrayList<>() : commandOptions;
```
Comment (NipunaRanasinghe) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java:114]:
```suggestion
        ArrayList<String> commandOptions = new ArrayList<>();
        if (args.get("commandOptions") instanceof ArrayList) {
            commandOptions = (ArrayList<String>) args.get("commandOptions");
        }
```

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java [hunk 4]
```diff
@@ -90,18 +95,29 @@
...
         command.addAll(commandOptions);
 
-        ArrayList<String> scriptArguments = (ArrayList<String>) args.get("scriptArguments");
+        ArrayList<String> scriptArguments = null;
+        if (args.get("commandOptions") instanceof ArrayList) {
+            scriptArguments = (ArrayList<String>) args.get("scriptArguments");
+        }
         scriptArguments = scriptArguments == null ? new ArrayList<>() : scriptArguments;
```
Comment (NipunaRanasinghe) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java:121]:
```suggestion
        ArrayList<String> scriptArguments = new ArrayList<>();
        if (args.get("commandOptions") instanceof ArrayList) {
            scriptArguments = (ArrayList<String>) args.get("scriptArguments");
        }
```

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java (hunk 3 above)
Comment (praveennadarajah) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java]: Fixed

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java (hunk 4 above)
Comment (praveennadarajah) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/launchrequest/LauncherImpl.java]: Fixed

**Pull request merged**